tests/bgp/test_traffic_shift.py

```diff
@@ -1,35 +1,52 @@
 import pytest
 import logging
 import ipaddr as ipaddress
-from bgp_helpers import parse_rib, verify_all_routes_announce_to_bgpmon 
+from bgp_helpers import parse_rib, verify_all_routes_announce_to_bgpmon,remove_bgp_neighbors,restore_bgp_neighbors
+from tests.common.helpers.constants import DEFAULT_ASIC_ID
 from tests.common.helpers.assertions import pytest_assert
 import re
 
 pytestmark = [
-    pytest.mark.topology('t1')
+    pytest.mark.topology('t1'),
+    pytest.mark.disable_loganalyzer
 ]
 
 logger = logging.getLogger(__name__)
 
 TS_NORMAL = "System Mode: Normal"
 TS_MAINTENANCE = "System Mode: Maintenance"
 TS_INCONSISTENT = "System Mode: Not consistent"
+TS_NO_NEIGHBORS = "System Mode: No external neighbors"
 
 @pytest.fixture
 def traffic_shift_community(duthost):
     community = duthost.shell('sonic-cfggen -y /etc/sonic/constants.yml -v constants.bgp.traffic_shift_community')['stdout']
     return community
 
+def verify_traffic_shift_per_asic(host, outputs, match_result, asic_index):
+    prefix = "BGP{} : ".format(asic_index) if asic_index != DEFAULT_ASIC_ID else ''
```

Comment:
TSA/B/C being system level feature why we want per asic validation ?

Reply:
This API was added to give flexibility to check the scenario of no neighbors per ASIC. With this support in a multi-asic DUT we need not do the loop and check for all ASIC's - as the behavior can be verified with on asic alone, here I use the first ASIC in single/multi-asic DUT.